airflow/models/dagrun.py

```diff
@@ -126,7 +150,7 @@ class DagRun(Base, LoggingMixin):
     creating_job_id = Column(Integer)
     external_trigger = Column(Boolean, default=True)
     run_type = Column(String(50), nullable=False)
-    conf = Column(PickleType)
+    _conf = Column(PickleType)
```

Comment:
This would require a database migration. I wonder if it would be better to make this a custom SQLAlchemy column type so selecting `conf` automatically converts to ConfDict without going through the additional `@property` layer. That would avoid possible bugs if someone does something like `session.scalar(select(DagRun.conf))` (this would not go through the `@property` getter!)

Reply:
Oh, boy. Important. Sorry did not catch this during review.

Reply:
@uranusjr Perhaps we could use hybrid_property here? 
https://docs.sqlalchemy.org/en/20/orm/mapped_attributes.html#:~:text=The%20approach%20above,extension%20as%20follows%3A